src/modules/vendor/components/programs/ProfileContent.vue

```diff
@@ -567,6 +591,21 @@ export default {
     isStepValid (step) {
       return step.areActivitiesValid && !step.activities.some(activity => activity.cards.length === 0);
     },
+    async openSubProgramPublicationModal () {
+      try {
+        const companies = await Companies.list();
+        this.companyOptions = this.formatCompanyOptions(companies);
+        this.subProgramPublicationModal = true;
+      } catch (e) {
+        console.error(e);
+        this.subProgramToPublish = null;
+      }
+    },
+    validateSubProgramPublicationModal (accessCompany) {
+      this.accessCompany = accessCompany;
+      this.subProgramPublicationModal = false;
+      this.openSubProgramPublicationConfirmationDialog();
```

Comment:
je pense que tu n'as pas besion de cette methode et tu peux mettre directement 

```
validateSubProgramPublicationModal (accessCompany) {
this.$q.dialog({})
.onOk(() => this.publishSubProgram(accessCompany))
.onCancel(() => NotifyPositive('Publication annulée.'));
}
```

Reply:
tu peux aussi enlever du coup la data `this.accessCompany` et ne passer al variable que en argument de fonction

Reply:
enfin je ne cacherais la modal des choix d'acces qu'une fois la validation passee et la publication effectuée sans erreur

Reply:
done
J'avais pensé à le faire mais je ne voulais pas passer un argument accessCompany à la fonction publish sachant que dans le cas d'un sous programme blended, on en a pas besoin.
Mais les deux solutions me vont